plutus-core/plutus-core/src/PlutusCore/Subst.hs

```diff
@@ -145,6 +148,64 @@ termSubstFreeNames
     -> Term tyname name uni fun ann
 termSubstFreeNames = purely termSubstFreeNamesA
 
+-- | Substitute a closed 'Type' for a type variable in a 'Type'. Does not descend under binders that
+-- bind the same variable as the one we're substituting for. The resulting 'Term' may and likely
+-- will not satisfy global uniqueness.
+typeSubstClosedType
```

Comment:
I just removed all `termSubstFreeNames` from the repo. They were kinda weird and we no longer need them, since they were only used for discharging in the CEK machine, which now works in a different way since the change to De Bruijn indices.